community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreAbstractNodeCursor.java

```diff
@@ -46,32 +47,50 @@ public StoreAbstractNodeCursor( NodeRecord nodeRecord, NodeStore nodeStore, Stor
     }
 
     @Override
-    public long getId()
+    public NodeItem get()
+    {
+        return this;
```

Comment:
This implementation doesn't check for illegal state. All (most at least?) other do. I think we can use `StatementConstants.NO_SUCH_NODE` to set as the node record id in the constructor and we'll just check that in get.